toolchain/lex/lex.cpp

```diff
@@ -895,10 +960,21 @@ auto Lexer::LexComment(llvm::StringRef source_text, ssize_t& position) -> void {
   bool is_valid_after_slashes = true;
   if (position + 2 < static_cast<ssize_t>(source_text.size()) &&
       LLVM_UNLIKELY(!IsSpace(source_text[position + 2]))) {
-    CARBON_DIAGNOSTIC(NoWhitespaceAfterCommentIntroducer, Error,
-                      "whitespace is required after '//'");
-    emitter_.Emit(source_text.begin() + position + 2,
-                  NoWhitespaceAfterCommentIntroducer);
+    llvm::StringRef comment_text = source_text.substr(position);
+    if (comment_text.starts_with("//@dump-sem-ir-start\n")) {
+      StartDumpSemIRRange(comment_text.begin());
+      AdvanceToLine(source_text, position, line_index_ + 1);
+      return;
+    } else if (comment_text.starts_with("//@dump-sem-ir-end\n")) {
```

Comment:
Does Carbon have a divergence from LLVM's "avoid else-after-return"? Otherwise perhaps drop the "else"s here.

Reply:
Carbon doesn't use LLVM style. I'm fine switching, just want to note it's not a "divergence from LLVM".

Reply:
Oh, right ,sorry - Google baseline, and it doesn't have this rule. No worries.